src/azure-cli-core/azure/cli/core/prompting.py

```diff
@@ -22,28 +23,64 @@ def _verify_is_a_tty():
         raise NoTTYException()
 
 
-def prompt(msg):
+def prompt(msg, help_string=None):
     _verify_is_a_tty()
-    return input(msg)
+    while True:
+        val = input(msg)
+        if val == '?' and help_string is not None:
+            print(help_string)
+            continue
+        return val
 
 
-def prompt_pass(msg='Password: ', confirm=False):
+def prompt_int(msg, help_string=None):
     _verify_is_a_tty()
-    password = getpass.getpass(msg)
-    if confirm:
-        password2 = getpass.getpass('Confirm ' + msg)
-        assert password == password2, 'Passwords do not match.'
-    return password
+
+    while True:
+        value = input(msg)
+        if value == '?' and help_string is not None:
+            print(help_string)
+            continue
+        try:
+            return int(value)
+        except ValueError:
+            print('{} is not a valid number'.format(value))
```

Comment:
done for the warnings, keeping print for the help string since that's not really a warning...